esp-hal/src/gpio/mod.rs

```diff
@@ -2024,7 +1946,7 @@ pub(crate) mod internal {
         }
 
         fn output_signals(&self, _: private::Internal) -> &[(AlternateFunction, OutputSignal)] {
-            handle_gpio_input!(&self.0, target, {
+            handle_gpio_output!(&self.0, target, {
```

Comment:
Even though input-only pins have no output signals, an explicit panic is more consistent with the rest of the driver.